applications/FluidDynamicsApplication/custom_elements/stokes_3D_twofluid.cpp

```diff
@@ -18,732 +18,750 @@ void Stokes3DTwoFluid::ComputeGaussPointLHSContribution(BoundedMatrix<double,16,
         const BoundedMatrix<double,nnodes,dim>& DN = data.DN_DX;
         const array_1d<double,nnodes>& N = data.N;
 
+        const double DN_0_0 = DN(0,0); const double DN_0_1 = DN(0,1); const double DN_0_2 = DN(0,2);
+        const double DN_1_0 = DN(1,0); const double DN_1_1 = DN(1,1); const double DN_1_2 = DN(1,2);
+        const double DN_2_0 = DN(2,0); const double DN_2_1 = DN(2,1); const double DN_2_2 = DN(2,2);
+        const double DN_3_0 = DN(3,0); const double DN_3_1 = DN(3,1); const double DN_3_2 = DN(3,2);
+
+        const double N_0 = N[0];
+        const double N_1 = N[1];
+        const double N_2 = N[2];
+        const double N_3 = N[3];
+
+        const double C_0_0 = C(0,0); const double C_0_1 = C(0,1); const double C_0_2 = C(0,2);
+        const double C_0_3 = C(0,3); const double C_0_4 = C(0,4); const double C_0_5 = C(0,5);
+        const double C_1_1 = C(1,1); const double C_1_2 = C(1,2); const double C_3_5 = C(3,5);
+        const double C_1_4 = C(1,4); const double C_1_5 = C(1,5); const double C_2_5 = C(2,5);
+        const double C_2_2 = C(2,2); const double C_2_3 = C(2,3); const double C_2_4 = C(2,4);
+        const double C_3_3 = C(3,3); const double C_4_4 = C(4,4); const double C_5_5 = C(5,5);
+        const double C_1_3 = C(1,3); const double C_3_4 = C(3,4); const double C_4_5 = C(4,5);
```

Comment:
Couldn't all these be references to avoid the allocation?

Reply:
Isnt it the same for doubles?

Reply:
You're right, it makes no difference in this case. Forget about my comment.